qiskit/opflow/zero_op.py

```diff
@@ -0,0 +1,119 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Zero Operator Class """
+
+from typing import Dict, List, Optional, Set, Union
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.opflow.list_ops.list_op import ListOp
+from qiskit.opflow.operator_base import OperatorBase
+from qiskit.opflow.primitive_ops.circuit_op import CircuitOp
+from qiskit.opflow.primitive_ops.matrix_op import MatrixOp
+from qiskit.opflow.primitive_ops.pauli_sum_op import PauliSumOp
+from qiskit.opflow.primitive_ops.primitive_op import PrimitiveOp
+from qiskit.opflow.state_fns.state_fn import StateFn
+from qiskit.quantum_info import Statevector
+from qiskit.utils.validation import validate_min
+
+
+class ZeroOp(OperatorBase):
```

Comment:
Is there a reason say why this is not a subclass of PrimitiveOp, as more of a fixed Primitive, rather than extending OperatorBase directly? Or is this somehow more general - I can partly understand since primitive  op construction is kinda complex already with the new

Reply:
I simply thought that StateFn and ListOp also had ZeroOp use cases. For example, I think it is necessary for SummedOp, but if there is no use case, it may be good to implement it in PrimitiveOp.